### PR TITLE
Auto deploy code to PyPi on git push

### PR DESCRIPTION
**Pull request by mstuttgart**
To this deploy command works, its need add the `PYPI_PASSWORD` environment variable in your TRavisCI settings panel. This variable store you PyPI password.

Details in [TravisCI doc](https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings).

The deploy happens when we push a new *tag* to repository.

Resolve https://github.com/nirmankarta/PyBeacon/issues/10

### CODE REVIEW

**Comment by PrabhanshuAttri** [comment]
@mstuttgart Good work. Will review and merge.

File: .travis.yml [hunk 1]
```diff
@@ -16,4 +16,13 @@ before_install:
...
+script: py.test
+
+deploy:
+  provider: pypi
+  distributions: "sdist bdist_wheel"
+  user: nirmankarta
+  password: $PYPI_PASSWORD
+  on:
+    tags: true
+    repo: nirmankarta/PyBeacon
```
[the remark this answers is not in the document]
Comment (mstuttgart) [reply .travis.yml]: @PrabhanshuAttri this deploy run only tag is pushed to repository. I use it in my projects and works fine. 

I will make this changes.

[the remark this answers is not in the document]
Comment (mstuttgart) [reply .travis.yml]: @PrabhanshuAttri this command:

```bash
IFS="'" read _ version_found _ <<< $(echo "$a" | sed '3q;d' PyBeacon/__init__.py)
```
not work. It raise this message:

```
Syntax error: redirection unexpected
``` 
I'm a begginer in shell script. Any idea?

[the remark this answers is not in the document]
Comment (PrabhanshuAttri) [reply .travis.yml]: @mstuttgart Let me check. I just wrote it without testing it.

[the remark this answers is not in the document]
Comment (PrabhanshuAttri) [reply .travis.yml]: In PyBeacon directory, just run the following commands
```
IFS="'" read _ version_found _ <<< $(echo "$a" | sed '3q;d' PyBeacon/__init__.py)
echo "$version_found"
```

[the remark this answers is not in the document]
Comment (PrabhanshuAttri) [reply .travis.yml]: I understood your way of using tags for deployment. I just wanted to make it automated without worrying much. Moreover, this way we can learn another way to deploy.

[the remark this answers is not in the document]
Comment (mstuttgart) [reply .travis.yml]: I understood, no problem! This tag way is used in other projects to. Its a nice way to contol release versions, but I will try make the shell script work.

[the remark this answers is not in the document]
Comment (mstuttgart) [reply .travis.yml]: TIps: I use [git hubflow](https://gist.github.com/kjunggithub/8330157) to create branchs and release branchs. When a done the work in release branch, the tool merge the release branch content and master and develop branchs and create tags for me. Then, the TravisCI see the tag and deploy to PyPI

[the remark this answers is not in the document]
Comment (PrabhanshuAttri) [reply .travis.yml]: Let's try this.

```
before_deploy:
function version_gt() { test "$(printf '%s\n' "$@" | sort -V | head -n 1)" != "$1"; } \
&& IFS="'" read _ version _ <<< $(echo "$a" | sed '3q;d' PyBeacon/__init__.py) && export VERSION_FOUND=$(echo $version | awk -F"'" '{print $1}') SHOULD_DEPLOY=0 \
&& echo "Current version: ${PB_VERSION} and version found: ${VERSION_FOUND}" \
&& if version_gt $VERSION_FOUND $PB_VERSION; then echo "Latest version ${VERSION_FOUND} will be deployed" && export SHOULD_DEPLOY=1 PB_VERSION=$(echo "${VERSION_FOUND}"); fi
```

```
deploy on condition:
$SHOULD_DEPLOY
```

**Comment by mstuttgart** [comment]
@PrabhanshuAttri .travis.yml update with you request changes. :) Please, take a look.

**Comment by PrabhanshuAttri** [comment]
Hi @mstuttgart I think you have missed my [last comment](https://github.com/nirmankarta/PyBeacon/pull/44#discussion_r143334850) by mistake. I have fixed the issue you were facing. After editing, travis.yml would look like this. 

```
before_deploy:
  - function version_gt() { test "$(printf '%s\n' "$@" | sort -V | head -n 1)" != "$1"; } && IFS="'" read _ version _ <<< $(echo "$a" | sed '3q;d' PyBeacon/__init__.py) && export VERSION_FOUND=$(echo $version | awk -F"'" '{print $1}') SHOULD_DEPLOY=0 && echo "Current version: ${PB_VERSION} and version found: ${VERSION_FOUND}" && if version_gt $VERSION_FOUND $PB_VERSION; then echo "Latest version ${VERSION_FOUND} will be deployed" && export SHOULD_DEPLOY=1 PB_VERSION=$(echo "${VERSION_FOUND}"); fi

deploy:
  provider: pypi
  distributions: "sdist bdist_wheel"
  user: nirmankarta
  password: $PYPI_PASSWORD
  on:
    branch: master
    condition: $SHOULD_DEPLOY

```

Just remove `after_deploy`. That's not required. Thanks

**Comment by mstuttgart** [comment]
@PrabhanshuAttri sorry, my mistake! Fixed

**Comment by PrabhanshuAttri** [comment]
Cool. Please remove line 33 and 34 as well.

**Comment by mstuttgart** [comment]
Done :)

**Comment by PrabhanshuAttri** [comment]
Let's test it.

**Pull request merged**